gcc/rust/resolve/rust-name-resolver.h

```diff
@@ -321,14 +327,25 @@ class Scope
   void append_reference_for_def (NodeId refId, NodeId defId)
   {
     bool ok = false;
-    iterate ([&] (Rib *r) mutable -> bool {
-      if (r->decl_was_declared_here (defId))
-	{
-	  ok = true;
-	  r->append_reference_for_def (defId, refId);
-	}
-      return true;
-    });
+
```

Comment:
This is where the defId is looked up only in the type NS and where it should probably use the same search behavior (ie. search in modules if not found in type NS).